sentry_sdk/client.py

```diff
@@ -510,7 +510,7 @@ def capture_event(
         dynamic_sampling_context = (
             event_opt.get("contexts", {})
             .get("trace", {})
-            .pop("dynamic_sampling_context", {})
+            .get("dynamic_sampling_context", {})
```

Comment:
you kinda need to pop it here because we don't actually want it to be part of `contexts.trace`, this is a hack (for reasons... :/) to get it and put it on the envelope headers `headers["trace"]` below.

https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#envelope-header